metricbeat/scripts/fields_collector.py

```diff
@@ -0,0 +1,51 @@
+import os
+
+# Collects fields for all modules and metricset
+
+
+def collect():
+
+    base_dir = "module"
+    path = os.path.abspath("module")
+
+    # yml file
+    fields_yml = ""
+
+    # Iterate over all modules
+    for module in os.listdir(base_dir):
+
+        module_fields = path + "/" + module + "/fields.yml"
+
+        # Only check folders where fields.yml exists
+        if os.path.isfile(module_fields) == False:
+            continue
+
+        # Load module yaml
+        with file(module_fields) as f:
+            tmp = f.read()
+            fields_yml += tmp
+
+
+        # Iterate over all metricsets
+        for metricset in os.listdir(base_dir + "/" + module):
+
+            metricset_fields = path + "/" + module + "/" + metricset + "/fields.yml"
+
+            # Only check folders where fields.yml exists
+            if os.path.isfile(metricset_fields) == False:
+                continue
+
+            # Load metricset yaml
+            with file(metricset_fields) as f:
+                # Add 4 spaces for indentation in fron of each line
+                for line in f:
+                    fields_yml += "    " + line
```

Comment:
This will insert trailing whitespace when the `line` contains only a newline. Consider adding a check for this condition or just removing all trailing whitespace from `fields_yml`.